src/phone-number-generator.ts

```diff
@@ -13,15 +13,14 @@ export default function generatePhoneNumber(
       getCountryPhoneDataByConfig(config);
     for (let j = 0; j < 100; j++) {
       const phoneNumber = getRandomPhoneNumber(countryPhoneData, config?.withoutCountryCode);
-      if (config?.withoutCountryCode ? isValidNumberForRegion(phoneNumber, countryPhoneData.alpha2 as CountryCode) : isPhoneNumber(phoneNumber)) {
+      if (config?.withoutCountryCode ? isValidNumberForRegion(phoneNumber, countryPhoneData.alpha2 as CountryCode) : (isValidPhoneNumber(phoneNumber) && isPhoneNumber(phoneNumber))) {
```

Comment:
Let's create a validation function for those 2 checks.
In this way we could add tests for that